src/WebJobs.Script/Description/DotNet/DotNetFunctionInvoker.cs

```diff
@@ -326,6 +339,27 @@ private MethodInfo CreateFunctionTarget(CancellationToken cancellationToken)
             }
         }
 
+        private async Task VerifyPackageReferencesAsync()
+        {
+            try
+            {
+                if (Metadata.IsAutoPackageRestoreEnabled && _metadataResolver.RequiresPackageRestore(Metadata))
```

Comment:
This logic checks our newly introduced flag and only verifies the need to run a package restore if its value is `true`